natiami/assignment1/Q1.cpp

```diff
@@ -0,0 +1,48 @@
+#include<iostream>
+#include<string>
+#include<vector>
+#include<fstream>
+#include <algorithm>
+#include <array> 
+
+using namespace std;
+
+bool checkAnagrams(string str1, string str2, bool caseSensitive){
+	vector<int> letters(52, 0); // array for letters
+
+	if (str1.length() != str2.length()) return false; // if size differs return false right away
+
+	for (size_t i = 0; i < str1.length(); i++)
+	{
+		letters[int(str1[i]) - 65]++;
+		letters[int(str2[i]) - 65]--;
+	}
+
+	// If algorithm is not case sensitive
+	if (!caseSensitive){
+		for (size_t i = 0; i < letters.size()/2; i++)
+			if (letters[i] != 0 && letters[i + 32] == 0) return false;

```

Comment:
This should be opposite. I was trying to convert to lower case letter not the upper one, since 'A' integer value is 65 and 'a' - 97, so the difference between any upper and lower case letters is 32